### ISSUE TITLE
_MultiThreadedRendezvous is a subclass of grpc.RpcError

### ISSUE THREAD

**Issue by dylanmorroll** [grpc/grpc#25334]
### What version of gRPC and what language are you using?
1.31.0

### What operating system (Linux, Windows,...) and version?
Linux Debian Buster

### What runtime / compiler are you using (e.g. python version or version of gcc)
Python 3.8

### What did you do?
We have an authentication interceptor that deals with some responses. In there, we have some code akin to the following:
```
if isinstance(response, grpc.RpcError):
    raise response
else:
    return response
````

When debugging I found the following:
```
>>> isinstance(response, grpc.RpcError)
True
>>> issubclass(response.__class__, grpc.RpcError)
True
>>> response.__class__
<class 'grpc._channel._MultiThreadedRendezvous'>
```

So due to _MultiThreadedRendezvous being a subclass of RpcError, my response was being raised as an exception not returned.

Which was giving me this in my logs:
```
    raise response
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.OK
	details = ""
>
```

### What did you expect to see?
I would have expected a valid response to be returned so then it could be handled by the rest of software correctly.

### What did you see instead?
This seemingly valid response was raised as an error, and due to it being a subclass of an `Exception` type, it was syntactically correct. Took me a while to figure out what was causing me the issue.

Am I missing something here? Is there a valid reason for _MultiThreadedRendezvous being an Exception type? To me it doesn't seem right that you can raise a valid response but I'd be very interested to know more!

### Anything else we should know about your project / environment?
Nope.

Cheers,
Dylan.

**Comment by lidizheng** [comment]
Thanks for filing this issue, the structure of classes for invoking-side does confuse people. What you observed is right, `_MultiThreadedRendezvous` (see [code](https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/_channel.py#L630)) is implementing 3 interfaces `grpc.Future`, `grpc.Call`, and `grpc.RpcError`. And, when an error occurs, gRPC Python raises the "Rendezvous" object as `grpc.RpcError` but expects users to use it as [`grpc.Call`](https://grpc.github.io/grpc/python/grpc.html#grpc.Call).

Unfortunately, changing this would be an API breaking change, because this design was released at 1.0. There is a proposal for bumping the major version, and dealing with the exception design: [gRFC-L76](https://github.com/grpc/proposal/pull/217). Feel free to leave your thoughts.

**Comment by dylanmorroll** [comment]
Thanks for linking the proposal, it sounds interesting :)

Glad to hear I wasn't just completely missing something there!
